Wrappers/Python/ccpi/web_viewer/trame_viewer.py

```diff
@@ -0,0 +1,772 @@
+            thumb_label=True,
+            style=style
+        )
+
+    def construct_slice_level_slider(self):
+        if self.cmax > 100:
+            # Use actual value
+            min_value = self.cmin
+            max_value = self.cmax
+            step = 1
+            self.slice_level_slider_is_percentage = False
+        else:
+            # Use percentages
+            min_value = 0
+            max_value = 100
+            step = 0.5
+            self.slice_level_slider_is_percentage = True
+
+        if self.slice_window_sliders_are_detailed:
+            style = "max-width: 300px"
+        else:
+            style = "visibility: hidden; height: 0"
+
+        return vuetify.VSlider(
+            v_model=("slice_level", self.slice_level_default),
+            min=min_value,
+            max=max_value,
+            step=step,
+            hide_details=True,
+            dense=True,
+            disabled=self.disable_2d,
+            label="Slice level",
+            thumb_label=True,
+            style=style
+        )
+
+    def construct_slice_window_range_slider(self):
+        if self.cmax > 100:
+            # Use actual values
+            min_value = self.cmin
+            max_value = self.cmax
+            step = 1
+            self.slice_window_slider_is_percentage = False
+        else:
+            # Use percentages
+            min_value = 0
+            max_value = 100
+            step = 0.5
+            self.slice_window_slider_is_percentage = True
+
+        if not self.slice_window_sliders_are_detailed:
+            style = "max-width: 300px"
+        else:
+            style = "visibility: hidden; height: 0"
+
+        return vuetify.VRangeSlider(
+            v_model=("slice_window_range", self.slice_window_range_defaults),
+            min=min_value,
+            max=max_value,
+            step=step,
+            hide_details=True,
+            dense=True,
+            disabled=self.disable_2d,
+            label="Slice window",
+            thumb_label=True,
+            style=style,
+        )
+
+    def construct_colour_slider(self):
+        if self.cmax > 100:
+            # Use actual values
+            min_value = self.cmin
+            max_value = self.cmax
+            step = 1
+            self.colour_slider_is_percentage = False
+        else:
+            # Use percentages
+            min_value = 0
+            max_value = 100
+            step = 0.5
+            self.colour_slider_is_percentage = True
+        
+        return vuetify.VRangeSlider(
+            label="Colour range",
+            hide_details=True,
+            solo=True,
+            v_model=("colouring", self.windowing_defaults),
+            min=min_value,
+            max=max_value,
+            step=step,
+            disabled=self.disable_3d,
+            thumb_label=True,
+            style="max-width: 300px"
+        )
+
+    def construct_windowing_slider(self):
+        if self.cmax > 100:
```

Comment:
If below 100 the slider would represent less than 100 individual steps, so we switched to percentages in this instance, specifically the big issue with it was .nxs files that seemed to return the data with max less than 1.

Reply:
Ah yes the nsx file has been normalised to be between 0 and 1 and I guess that makes the numbers on the slider messy if you use the raw values.
As a future thing, it would be nice if the interface can indicate when percentages are used rather than raw values so that the user is aware